interface-impl-amf/src/main/java/org/mule/amf/impl/model/ParameterImpl.java

```diff
@@ -248,4 +260,14 @@ private static boolean hasAnArrayVariant(UnionShape unionShape) {
 
     return hasAnArrayVariant;
   }
+
+  private static boolean hasNilShape(UnionShape unionShape) {
+    for (Shape shape : unionShape.anyOf()) {
+      if (shape instanceof NilShape) {
+        return true;
+      }
+    }
+    return false;
```

Comment:
With the Streams API you can also write it like this:
```suggestion
    return unionShape.anyOf().stream().anyMatch(NilShape.class::isInstance);
```
(imho it's just a matter of preference, probably the current version is more approachable and easy to understand)